pkg/reconciler/shared/tektonconfig/tektonconfig.go

```diff
@@ -138,3 +152,64 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, tc *v1alpha1.TektonConfi
 
 	return nil
 }
+
+func (r *Reconciler) createOperatorVersionConfigMap(tc *v1alpha1.TektonConfig) error {
+	koDataDir := os.Getenv(common.KoEnvKey)
+	operatorDir := filepath.Join(koDataDir, "info")
+
+	if err := common.AppendManifest(&r.manifest, operatorDir); err != nil {
+		return err
+	}
+
+	manifest, err := r.manifest.Transform(
+		mf.InjectNamespace(tc.GetSpec().GetTargetNamespace()),
+	)
+	if err != nil {
+		return err
+	}
+
+	if err = manifest.Apply(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *Reconciler) ensureTargetNamespaceExists(ctx context.Context, tc *v1alpha1.TektonConfig) error {
+
+	nps, err := r.kubeClientSet.CoreV1().Namespaces().List(ctx, metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("operator.tekton.dev/targetNamespace=%s", "true"),
+	})
+
+	if err != nil {
+		return err
+	}
+
+	if len(nps.Items) > 0 {
+		for _, j := range nps.Items {
+			if j.Name != tc.GetSpec().GetTargetNamespace() {
+				if err := r.kubeClientSet.CoreV1().Namespaces().Delete(ctx, tc.GetSpec().GetTargetNamespace(), metav1.DeleteOptions{}); err != nil {
+					return err
+				}
```

Comment:
Functionaly the logic here is correct. However, this looks a bit agressive. I mean, here we are deleting previous targetNamespace(s) before components get a chance to properly finalize/move items in their respective reconcilers.

My suggestion is to split what this function does into 2 funcitons.

1. In this `ensureTargetNamespaceExists` function instead of deleting the previous targetNamespaces add an additional label to them `"operator.tekton.dev/targetNamespace/mark-for-deletion": "true"`
2. Write a second function `deleteObsoleteTargetNamespaces()`
     Then query namespaces with `"operator.tekton.dev/targetNamespace/mark-for-deletion": "true"` and then loop and delete them. 
    👉 call `deleteObsoleteTargetNamespaces()` at the end of TektonConfig reconciler just before the final update call ([here](https://github.com/tektoncd/operator/blob/2217ee6cdf0fb31842459a7b339d88fcbf8d5049/pkg/reconciler/shared/tektonconfig/tektonconfig.go#L133)) . 
Note: if there is an error in deleting ohsolete namespaces, then let us log it and continue, let us not add a status field for that now.